docs/en/operations/settings/settings.md

```diff
@@ -404,6 +404,26 @@ Possible values:
 
 Default value: 0.
 
+## any_join_distinct_right_table_keys {#any_join_distinct_right_table_keys}
+
+Enables or disables specific ClickHouse server behavior in `ANY JOIN` operations.
+
+When the behavior enabled, ClickHouse uses the logic with many-to-one left-to-right table keys mapping in `ANY JOIN` operations. This leads to confusing not equal results for queries with `t1 ANY LEFT JOIN t2` and `t2 ANY RIGHT JOIN t1` constructions.
+
+When the behavior disabled, ClickHouse uses the logic which provides one-to-many keys mapping in `ANY RIGHT JOIN` operations. This guaranties equality of results for queries with `t1 ANY LEFT JOIN t2` and `t2 ANY RIGHT JOIN t1` constructions.
```

Comment:
It change behaviour for `ANY INNER JOIN` as well. If setting is enabled you'll get all rows from left table (same behaviour as in `SEMI LEFT JOIN`). If it's disabled you'll one row per key from both left and right tables.